### PR TITLE
Migrate library to use module: "NodeNext" as recommended by TypeScript and SvelteKit docs

### PR DESCRIPTION
**Pull request by rsweeneydev**
Fixes: #909

### CODE REVIEW

**Comment by rsweeneydev** [comment]
Here's more detail on why this change is necessary. The tldr is that TypeScript libraries should use `"module": "NodeNext"` to ensure they are the most compatible. NodeNext enforces the strictest possible settings and will enable the library to be used in node.js, the browser, or a bundler.

I'm intentionally overriding both`module` and `moduleResolution` to align the library with the TypeScript and SvelteKit best practices for library authors. `module: NodeNext` must be used with `moduleResolution: NodeNext`. Since both `module` and `moduleResolution` are set in .svelte-kit/tsconfig.json, we have to override both.

Here's more detail on why libraries should be NodeNext. From the TypeScript handbook:

> When compiling an app, you choose the moduleResolution option for a TypeScript project based on who the module resolution [host](https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution-is-host-defined) is. When compiling a library, you don’t know where the output code will run, but you’d like it to run in as many places as possible. Using "module": "nodenext" (along with the implied ["moduleResolution": "nodenext"](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext)) is the best bet for maximizing the compatibility of the output JavaScript’s module specifiers, since it will force you to comply with Node.js’s stricter rules for import module resolution.
>
>...
>
> In short, "moduleResolution": "bundler" is infectious, allowing code that only works in bundlers to be produced. Likewise, "moduleResolution": "nodenext" is only checking that the output works in Node.js, but in most cases, module code that works in Node.js will work in other runtimes and in bundlers.

https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution-for-libraries

They are a little more direct in the reference section of the TypeScript docs.

> A common misconception is that node16 and nodenext only emit ES modules. In reality, node16 and nodenext describe versions of Node.js that support ES modules, not just projects that use ES modules. Both ESM and CommonJS emit are supported, based on the [detected module format](https://www.typescriptlang.org/docs/handbook/modules/reference.html#module-format-detection) of each file. Because node16 and nodenext are the only module options that reflect the complexities of Node.js’s dual module system, **they are the only correct module options for all apps and libraries that are intended to run in Node.js v12 or later, whether they use ES modules or not.**

https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext

It doesn't seem possible to combine `module: esnext` with `moduleResolution: NodeNext`.

> --moduleResolution node16 and nodenext must be paired with their [corresponding module value](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext).

https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1

The inverse is also true. The bundler option only works with esnext.

> --moduleResolution bundler must be paired with the --module esnext option.

And finally, in the Guide, there is a section titled: "[I'm writing a library](https://www.typescriptlang.org/docs/handbook/modules/guides/choosing-compiler-options.html#im-writing-a-library)", it says to use `module: node16` and explains why this is the best choice for compatibility. Node16 = NodeNext

**Review by TGlide — COMMENTED** [review]
Thank you so much!

File: package.json
Comment (AdrianGonz97) [inline package.json:1]: `@sveltejs/package` needs to be updated to the latest for the same reason provided in the Bits PR: https://github.com/huntabyte/bits-ui/pull/277#discussion_r1451766367

**Review by AdrianGonz97 — CHANGES_REQUESTED** [review]

File: package.json
Comment (rsweeneydev) [reply package.json:1]: Thanks for catching this. I upgraded @sveltejs/package to 2.2.5 with the latest commit.

**Comment by TGlide** [comment]
Some type errors that'll probably get fixed with #918. So will wait on that for now

**Pull request merged**